script/release_utils.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+set -e
+
+function commit_list {
+  local tag=${1:?}
+  local repo_domain=${2:?}
+  local repo_name=${3:?}
+  git fetch --tags
+  local previous_tag=`curl -H "Authorization: token $ACCESS_TOKEN" -s https://api.github.com/repos/$repo_domain/$repo_name/tags | jq --raw-output '.[1].name'`
+  local commit_list=`git log $previous_tag..$tag --oneline`
+  echo "$commit_list" | sed -n -e 'H;${x;s/\n/\n- /g;p;}'
+}
+
+function get_release_notes {
+  local tag=${1:?}
+  local repo_domain=${2:?}
+  local repo_name=${3:?}
+  local commits=`commit_list $tag $repo_domain $repo_name`
+  local current_dir=`cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd`
+  local release_notes=`echo "$(cat $current_dir/release_notes.tpl)${commits}" | sed 's/<<TAG>>/'"${tag}"'/g'`
+  local escaped_release_notes=`echo "$release_notes" | sed -n -e 'H;${x;s/\n/\\\n/g;s/"/\\"/g;p;}'`
+  echo "${escaped_release_notes}"
+}
+
+function get_release_body {
+  local tag=${1:?}
+  local repo_domain=${2:?}
+  local repo_name=${3:?}
+  local release_notes=$(get_release_notes $tag $repo_domain $repo_name)
+  echo '{
+    "tag_name": "'$tag'",
+    "target_commitish": "master",
+    "name": "'$tag'",
+    "body": "'$release_notes'",
+    "draft": true,
+    "prerelease": false
+  }'
+}
+
+function update_release_tag {
+  local tag=${1:?}
+  local repo_domain=${2:?}
+  local repo_name=${3:?}
+  local release_id=$(curl -H "Authorization: token $ACCESS_TOKEN" -s https://api.github.com/repos/$repo_domain/$repo_name/releases | jq  --raw-output '.[0].id')
+  local body=$(get_release_body $tag $repo_domain $repo_name)
+  local release=`curl -H "Authorization: token $ACCESS_TOKEN" -s --request PATCH --data $body  https://api.github.com/repos/$repo_domain/$repo_name/releases/$release_id`
+  echo $release
+}
+
+function release_tag {
+  local tag=$1
+  local repo_domain=${2:?}
+  local repo_name=${3:?}
+  local body=$(get_release_body $tag $repo_domain $repo_name)
+  local release=`curl -H "Authorization: token $ACCESS_TOKEN" -s --request POST --data "$body" https://api.github.com/repos/$repo_domain/$repo_name/releases`
+  echo $release
+}
+
+function upload_asset {
```

Comment:
Any reason we don't use the recommended `ghr` tool? https://circleci.com/blog/publishing-to-github-releases-via-circleci/

Reply:
I tried with that tool at the beginning but it always create an empty release with the files uploaded. It doesn't allow you to mark it as a draft or to add release notes, that's why I use directly the API.

Reply:
@andresmgot ah interesting, thanks